telemetry/prometheus/src/test/scala/io/buoyant/telemetry/prometheus/PrometheusTelemeterTest.scala

```diff
@@ -144,4 +144,68 @@ class PrometheusTelemeterTest extends FunSuite {
     val rsp = await(handler(Request(prometheusPath))).contentString
     assert(rsp == "linkerd_rt:server:requests{rt=\"incoming\", server=\"127.0.0.1/4141\"} 1\n")
   }
+
+  test("exception stats are labelled for service") {
+    val (stats, handler) = statsAndHandler
+    // Replicate stack trace for some failure bar:baz
+    stats.scope("rt", "incoming", "service", "/#/foo", "failures").counter("bar").incr()
+    stats.scope("rt", "incoming", "service", "/#/foo", "failures").counter("bar:baz").incr()
```

Comment:
Related to my question above, is this how these types of nested exceptions are actually recorded?  Or would it be more like `stats.scope("rt", "incoming", "service", "/#/foo", "failures", "bar").counter("baz").incr()` ?

Reply:
Also related to my answer above, this is how I've seen them recorded. If you have seen differently then I would be interested in seeing that!

Reply:
@kleimkuhler I think I understand what @adleong means when referring to how nested exceptions are recorded. I took a look at how the failure stats get generated for a Finagle HTTPServer. Check out this [code](https://github.com/twitter/finagle/blob/63d5858f0a7d5f926ec8107233bf95d42c6fcda7/finagle-base-http/src/main/scala/com/twitter/finagle/http/codec/HttpServerDispatcher.scala#L47) in the Finagle code base. It looks like counter receives a list of exception class name strings and then is incremented. That means that your test would have something like:
```scala
stats.scope("rt", "incoming", "service", "/#/foo", "failures").counter("Exception1", "Exception2", "Exception3").incr()
```

Reply:
@dadjeibaah Okay thanks for the link - I understand what you mean. I can work on adding a catch for multiple exceptions within one counter.